evap/results/tools.py

```diff
@@ -202,6 +202,19 @@ def average_non_grade_rating_questions_distribution(results):
     )
 
 
+def calculate_average_course_distribution(course):
+    if not all(evaluation.state == 'published' for evaluation in course.evaluations.all()):
```

Comment:
"if any not equal" instead of "if not all equal"?
Also, maybe delegate to database again?